src/librbd/io/ImageRequest.cc

```diff
@@ -407,6 +410,27 @@ void ImageReadRequest<I>::send_image_cache_request() {
                                   req_comp);
 }
 
+template <typename I>
+bool AbstractImageWriteRequest<I>::finish_request_early() {
+  AioCompletion *aio_comp = this->m_aio_comp;
+  {
+    std::shared_lock image_locker{this->m_image_ctx.image_lock};
+    if (this->m_image_ctx.snap_id != CEPH_NOSNAP || this->m_image_ctx.read_only) {
+      aio_comp->fail(-EROFS);
+      return true;
+    }
+  }
+  uint64_t total_bytes = 0;
+  for (auto& image_extent : this->m_image_extents) {
+    total_bytes += image_extent.second;
+  }
+  if (!total_bytes) {
```

Comment:
Nit: (it's not a boolean)

```suggestion
  if (total_bytes == 0) {
```

Reply:
Updated.